### DISCUSSION TITLE
renderer not changing/using wrong renderer

### DISCUSSION THREAD

**Discussion by dongfanglong2012** [r0x0r/pywebview#827] [Q&A]
I have a project that after doing some testing seems to always load using the edgehtml gui on my other computer. This breaks the scrollbars and a in general makes things not work right. On the computer I made it on it uses the right one though. I tried using `webview.start(api.set_window, window, gui='edgechromium')` per the web engine page but it doesn't seem to matter. when i take it to the other computer it uses the edgehtml one. If i load the .html file in a edge or chrome browser it loads right. But for some reason it seems on my other computer the gui flag doesn't seem tot do anything. Am I missing a setting somewhere? I have tried all of them and it seems no matter what one i set it too on the other computer it uses edgehtml.

**Comment by r0x0r** [comment]
This means that edgechromium is not available on your system and it falls back to edgehtml. You have to have Edge Webview2 runtime installed first.

**Reply by dongfanglong2012** [reply]
thats strange i thought it was part of edge now. If i just load the html in edge it works so im guessing its only inside the browser then

**Reply by dongfanglong2012** [reply]
that doesnt seem to be the problem. i went and manually installed it but there is no change

**Reply by dongfanglong2012** [reply]
i thought since it was only doing it on that one computer that maybe it was the .net but it has version 4.8 and the edgechromium  says it only needs 4.6.2. it has the right .net and it has the webview 2 runtime but for some reason when the page is loaded in the app it is clearly using the wrong renderer.

**Comment by r0x0r** [comment]
https://github.com/r0x0r/pywebview/blob/7641414db75542958d1d1158b903576aafebd47c/webview/platforms/winforms.py#L65
Could you debug this code and see what it returns and what dependencies it is missing? 

Did you install this runtime?
https://developer.microsoft.com/en-us/microsoft-edge/webview2/#download-section

**Reply by dongfanglong2012** [reply]
yes that is installed on the computer in question. im looking at that py file you liked. have to try it out first
